### PR TITLE
Use new endpoint for running remote queries

### PR DESCRIPTION
**Pull request by shati-patel**
Depends on internal PR, which has now been merged 😄 

@robertbrignull added a new endpoint for triggering remote query runs, so we no longer use workflow dispatch.

## Checklist

N/A - internal only (no user-facing changes)

- ~[CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.~
- ~Issues have been created for any UI or other user-facing changes made by this pull request.~
- ~`@github/docs-content-codeql` has been cc'd in all issues for UI or other user-facing changes made by this pull request.~

### CODE REVIEW

**Review by robertbrignull — APPROVED** [review]
LGTM, and we tested this together when writing it. Up to you if you want to also get approval from someone else more familiar with the extension.

File: extensions/ql-vscode/src/run-remote-query.ts
Comment (aeisenberg) [inline extensions/ql-vscode/src/run-remote-query.ts:45]: Presumably when this is released, we will no longer be pointing to internal repositories?

It's not great that we have some code in this repo that is only usable with proper permissioning, but since this is guarded by a canary it's fine.

**Review by aeisenberg — APPROVED** [review]
This is fine by me.

File: extensions/ql-vscode/src/run-remote-query.ts
Comment (shati-patel) [reply extensions/ql-vscode/src/run-remote-query.ts:42]: > Will this API endpoint ever be made public?

Eventually, yes (but probably not for a long time 🙃)

**Pull request merged**
